### ISSUE TITLE
Slaves not being removed from DB (I am guessing - in presence of Imperial slaves)

### ISSUE THREAD

**Issue by Tromador** [eyeonus/Trade-Dangerous#25]
See the following excerpt from my TradeDangerous.prices. There are no slaves presently for sale at this station  (HR 8208/Sturgeon Station)

```
   + Slavery
      Imperial Slaves                 17152   17510          ?   139659M  2018-09-17 13:02:48
      Slaves                          11039   11650          ?    54801M  2018-09-15 18:03:56
```
My best guess is that because "Slaves" is a subset of "Imperial Slaves" it did not get deleted when the new station data was uploaded, you can see the disparity of date/time information.

**Comment by eyeonus** [comment]
Yay. I'll look into it.

**Comment by eyeonus** [comment]
So, the problem is, the plugin doesn't remove no longer existing items from the DB. I just hadn't implemented it.

I /think/ I have a solution, but I haven't tested it yet and I don't really have the time to do so right now.

I'm uploading it to the dev branch now, in case you want to test it. I'll push it to main once we know for sure it works.

**Comment by eyeonus** [comment]
Do an update, remove some entries from one or more of the stations in the source file and save it. Do a second update with the force option, see if the entries you removed from the source file are gone in the DB.

The code I added, aiming no bugs or unintended results, does the following:

Set current station to -1, create empty dict.

For each item in the source file:

Check if current station matches item's station. (Well be false for the first item.)

If not, delete all items in the current dict set to false, delete the dict, set current station to the new station, and create a dict of all the possible items set to false

If so, set the current item to true in the dict.

**Comment by eyeonus** [comment]
Fixed in 8a43404
